apps/dash-label-properties/app.py

```diff
@@ -0,0 +1,456 @@
+import dash
+from dash.dependencies import Input, Output, State
+import dash_html_components as html
+import dash_core_components as dcc
+import dash_bootstrap_components as dbc
+import dash_table
+from dash_table.Format import Format
+import plotly.express as px
+import plotly.graph_objs as go
+import datashader as ds
+
+import numpy as np
+from skimage import io, filters, measure
+import pandas as pd
+
+import PIL
+from skimage import color, img_as_ubyte
+from plotly import colors
+
+import base64
+import xarray as xr
+
+external_stylesheets = [dbc.themes.BOOTSTRAP, "assets/object_properties_style.css"]
+
+app = dash.Dash(__name__, external_stylesheets=external_stylesheets)
+server = app.server
+
+# Read the image that will be segmented
+filename = (
+    "https://upload.wikimedia.org/wikipedia/commons/a/ac/Monocyte_no_vacuoles.JPG"
+)
+img = io.imread(filename, as_gray=True)[:660:2, :800:2]
+label_array = measure.label(img < filters.threshold_otsu(img))
+current_labels = np.unique(label_array)[np.nonzero(np.unique(label_array))]
+# Compute and store properties of the labeled image
+prop_names = [
+    "label",
+    "area",
+    "perimeter",
+    "eccentricity",
+    "euler_number",
+    "mean_intensity",
+]
+prop_table = measure.regionprops_table(
+    label_array, intensity_image=img, properties=prop_names
+)
+table = pd.DataFrame(prop_table)
+# Format the Table columns
+columns = [
+    {"name": label_name, "id": label_name, "selectable": True}
+    if precision is None
+    else {
+        "name": label_name,
+        "id": label_name,
+        "type": "numeric",
+        "format": Format(precision=precision),
+        "selectable": True,
+    }
+    for label_name, precision in zip(prop_names, (None, None, 4, 4, None, 3))
+]
+# Select the columns that are selected when the app starts
+initial_columns = ["label", "area"]
+
+img = img_as_ubyte(color.gray2rgb(img))
+img = PIL.Image.fromarray(img)
+
+
+def image_with_contour(
+    img, active_labels, data_table, active_columns, color_column, mode="lines"
+):
+    """
+    Figure with contour plot of labels superimposed on background image.
+
```

Comment:
Since this function is quite advanced, would it be possible to give a few explanations in the docstring about the different traces which are used, and the tricks to improve performance?